DEPLOYMENT.md

```diff
@@ -0,0 +1,57 @@
+
+## RHMAP Proxy
+
+RHMAP Proxy is a separate component which should be deployed to OpenShift per MBaaS project. Its purpose is to route requests that are not using the standard URL scheme. The RHMAP Proxy application is bundled within the RHMAP RPM in the form of an OpenShift template. Creating this application will create an OpenShift route which will expose this service. Carry out the following steps to configure RHMAP Core to use non-wildcard routing.
+
+**_1.  Setting Millicore environment variable for client side connectivity_**
+
+Upon initialization, a client application must call out to RHMAP Core to discover the URL to which it should make API requests. We must set the following environment variable in Millicore to allow this to work behind RHMAP Proxy.
+
+```
+oc env dc millicore -n rhmap-core EXTERNAL_HOST=https://rhmaprouter.internal.domain.com
+```
+
+NOTE: The `EXTERNAL_HOST` value must match the value of the route which you will pass to the template for RHMAP Proxy
+
+
+**_2. Configure an MBaaS to use RHMAP Proxy_**
+
+When creating an MBaaS target, you must specify an external URL via the `”External MBaaS Host”` form field. The value provided here when creating the MBaaS target must match the value of `EXTERNAL_HOST` set previously.
+
+**_3. Deploy RHMAP Proxy_**
+
+The application takes a number of parameters which should be set in relation to your infrastructure:
+
+1. BASE_HOST - A **required** parameter which should be set to the domain name of the OpenShift router. This value should match the cluster variable [openshift_master_cluster_hostname](https://docs.openshift.com/container-platform/latest/install_config/install/advanced_install.html#configuring-cluster-variables)
+1. PLATFORM_URL - An optional parameter which sets the full URL to RHMAP Core Platform. Example `https://rhmap.internal.domain.com`. In the case of a single OpenShift infrastructure, this should be set and the value obtained by running the command `oc get route rhmap -n rhmap-core -o=yaml | grep host`
+1. EXTERNAL_ROUTE - An optional parameter which specifies explicitly the route that should be created by OpenShift to expose this component. If not provided, this will be auto generated by OpenShift. This should match the values provided in steps 1 and 2
+
+To deploy RHMAP Proxy run the following command passing the parameters described above as appropriate:
+
+```
+oc new-app -n <myMBaasProjectName> -f /opt/rhmap/templates/core/proxy/fh-nginx-proxy-template.json --param=BASE_HOST=internal.doamin.com --param=PLATFORM_URL=https://rhmap.internal.domain.com
+```
+
+### Other Considerations
+
+**_1. Deploying applications behind RHMAP Proxy_**
+
+If serving static content from your application, it is important to consider how paths are written within the applications source code. It is always advisable to use relative paths with dot notation. For example `<a href=”/contact>Contact Usa>` should be written as `<a href=”./contact>Contact Usa>` as appropriate.
```

Comment:
Probably worth giving an example of what the url will look like in the browser to make this clearer